default/python/AI/ColonisationAI.py

```diff
@@ -400,25 +400,16 @@ def get_colony_fleets():
         avail_pp_by_sys.update([(sys_id, available_pp[p_set]) for sys_id in set(PlanetUtilsAI.get_systems(p_set))])
     colony_cost = colony_pod_cost()
     outpost_cost = outpod_pod_cost()
-    production_queue = empire.productionQueue
-    queued_outpost_bases = []
-    queued_colony_bases = []
-    for queue_index in range(0, len(production_queue)):
-        element = production_queue[queue_index]
-        if element.buildType == EmpireProductionTypes.BT_SHIP and element.turnsLeft != -1:
-            if foAI.foAIstate.get_ship_role(element.designID) in [ShipRoleType.BASE_OUTPOST]:
-                build_planet = universe.getPlanet(element.locationID)
-                queued_outpost_bases.append(build_planet.systemID)
-            elif foAI.foAIstate.get_ship_role(element.designID) in [ShipRoleType.BASE_COLONISATION]:
-                build_planet = universe.getPlanet(element.locationID)
-                queued_colony_bases.append(build_planet.systemID)
+
+    colony_base_manager = OrbitalColonyBaseManager()
```

Comment:
I see only 2 usages of this class.  First one have side effects (clear production queue) and second one only mutates inner state (and have side effects).   

Can we change inner implementation in next way:

- make a copy of outer data
- process data
- update outer data
 
This will allow for the manager to be independent.   

or

 instead of passing `foAI.foAIstate.qualifyingColonyBaseTargets` to manager make foAI.foAIstate.qualifyingColonyBaseTargets to be a manager instance.